### PR TITLE
Clean up non failing test errors

### PR DESCRIPTION
**Pull request by islemaster**
Using the new `throwOnConsoleErrors` utility (#9595) I found a few apps tests that were passing when they probably should have been failing.  This PR made those tests fail, then fixes them.
- CompletionButtonTest
  - Failed because required prop `playspacePhoneFrame` was not provided.  This prop is a boolean flag.  I verified that it doesn't have to be required from the way it's used, then made it not required.
- instructionsComponentsTest
  - Failed because required prop `noInstructionsWhenCollapsed` was misspelled `noInstrucstionsWhenCollapsed`.  Fixed the typo.
  - Then failed because required prop `hasInlineImages` was not provided.  This prop is a boolean flag.  I verified that it doesn't have to be required from the way it's used, then made it not required.
- Level Tests:
  - ./gamelab/custom.js
    - Failed because propType `instructionsInTopPane` was not a function.  Not sure how this got missed originally, but changed to a simple boolean type.
  - ./applab/cspunit3.js
    - Failed because required prop `hasContainedLevels` was not provided.  This prop is a boolean flag.  I verified that it doesn't have to be required from the way it's used, then made it not required.

I went ahead and enforced that _no_ level tests should call `console.error()`.

I also noticed that level tests were being listed as `{testname}.js.js` so I fixed the doubled extension.

### CODE REVIEW

File: apps/src/gamelab/GameLabVisualizationColumn.jsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ var GameLabVisualizationColumn = function (props) {
 
 GameLabVisualizationColumn.propTypes = {
   finishButton: React.PropTypes.bool.isRequired,
-  instructionsInTopPane: BelowVisualization.propTypes.instructionsInTopPane,
+  instructionsInTopPane: React.PropTypes.bool.isRequired
```
Comment (islemaster) [inline apps/src/gamelab/GameLabVisualizationColumn.jsx:55]: @pcardune git blames you for this one :point_up:

File: apps/src/templates/instructions/MarkdownInstructions.jsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ const MarkdownInstructions = React.createClass({
   propTypes: {
     renderedMarkdown: React.PropTypes.string.isRequired,
     noInstructionsWhenCollapsed: React.PropTypes.bool.isRequired,
-    hasInlineImages: React.PropTypes.bool.isRequired,
+    hasInlineImages: React.PropTypes.bool,
```
Comment (islemaster) [inline apps/src/templates/instructions/MarkdownInstructions.jsx:31]: @Bjvanminnen FYI when you're back.

Comment (Hamms) [inline apps/src/templates/instructions/MarkdownInstructions.jsx:31]: this was actually my change

**Comment by Hamms** [comment]
Any reason why you only added `throwOnConsoleErrors` to a couple tests?

**Comment by Hamms** [comment]
This is awesome! LGTM

**Comment by islemaster** [comment]
I added it to all levelTests, and to the two unit tests that were logging errors.  I haven't added it to all tests because I think forcing tests to actually include this utility makes them less 'magic' and easier to understand.  I'm also unit-testing the `throwOnConsoleErrors` util itself; though we could write a one-off workaround for that test if you wanted to enable this behavior everywhere else.

We don't do this much, but I imagine we might want to verify appropriate use of console.error in some cases, and stubbing it with this utility precludes the ability to stub it differently for a particular test.

Do you think this is something we should enable on all JavaScript tests?

**Comment by Hamms** [comment]
Ah, I think my question was unclear. I meant to ask why you chose to add it specifically to those tests you added it to; I agree that adding it everywhere is likely unwise.

**Comment by islemaster** [comment]
I see.  I chose these tests by running the full suite and manually reviewing the log to find tests that were logging errors; when I got to level tests I realized that I _could_ wire up a way for certain level tests to include the util, but really I'm not sure any of them should be generating unexpected errors (and none of them currently do, after this PR).

**Pull request merged**